### PR TITLE
chore(gha): pin versions to hashes

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind cleanup

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area CI

**What this PR does / why we need it**:

We have started pinning some GitHub actions to their hash, especially the ones that were based on `master`. We should pin all of them. In order to limit potential breakage I have left out the actions that need a major version upgrade (except for checkout) so we can handle them in a subsequent PR.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by Andreagit97 — APPROVED** [review]
/approve

**Comment by LucaGuerra** [comment]
Seems like one upgrade is failing. I'm checking it.

File: .github/workflows/reusable_build_packages.yaml [hunk 1]
```diff
@@ -53,10 +53,11 @@ jobs:
           yum install -y wget git make m4 rpm-build perl-IPC-Cmd
     
       - name: Checkout
-        uses: actions/checkout@v3
+        # It is not possible to upgrade the checkout action to versions >= v4.0.0 because of incompatibilities with centos 7's libc.
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
```
Comment (LucaGuerra) [inline .github/workflows/reusable_build_packages.yaml:57]: this is the sad downgrade (actually non-upgrade) we have to do to be compatible with centos 7. I think it's ok.

**Review by FedeDP — APPROVED** [review]
/approve

**Review by leogr — APPROVED** [review]

**Pull request merged**